src/ui_basic_flows_nbgl.c

```diff
@@ -0,0 +1,251 @@
+#ifdef HAVE_NBGL
+#include <stdbool.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdint.h> 
+#include <os_io_seproxyhal.h>
+#include <ux.h>
+#include "ui_helpers.h"
+#include "glyphs.h"
+#include "nbgl_page.h"
+#include "nbgl_use_case.h"
+#include "menu.h"
+
+// Globals
+static ui_callback_fn_t * confirm_clbk;
+static ui_callback_fn_t * reject_clbk;
+static char* rejection_text;
+static nbgl_layoutTagValueList_t pairList;
+static nbgl_layoutTagValue_t fields[MAX_FIELDS_PER_TRANSACTION];
+static nbgl_pageInfoLongPress_t infoLongPress;
+static bool reviewStarted;
+static bool isDisplayAddress;
+static bool isDisplayPath;
+
+// Static functions declarations
+static void rejectUseCaseChoice(void);
+static void displayTransaction(void);
+static void rejectAndDisplayIdleMenu(void);
+static void displayAddress(void);
+
+// Static functions definitions
+static void warningChoice(bool accept) {
+    if (accept) {
+        confirm_clbk();
+    }
+    else {
+        rejectAndDisplayIdleMenu();
+    }
+}
+
+static void contractWarningChoice(bool accept) {
+    if (accept) {
+        displayTransaction();
+    }
+    else {
+        rejectAndDisplayIdleMenu();
+    }
+}
+
+
+static void reviewChoice(bool confirm) {
+  if (confirm) {
+    confirm_clbk();
+    nbgl_useCaseStatus("TRANSACTION\nCONFIRMED",true,ui_idle);
+  }
+  else {
+    rejectUseCaseChoice();
+  }
+}
```

Comment:
spacing is wildly inconsistent, adding a linter in a 2nd step would help a lot

Reply:
Sure will do.

Reply:
I have applied clang-format on the `basic_flows_nbgl` file. I'll add a PR for a new CI lint check when we have merged this one.